dags/inference/configs/jetstream_pytorch_gce_config.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct configs for maxtext inference DAG."""
+
+import datetime
+import json
+from typing import Dict
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags import test_owner
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion
+
+PROJECT_NAME = Project.CLOUD_ML_AUTO_SOLUTIONS.value
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+GCS_SUBFOLDER_PREFIX = test_owner.Team.INFERENCE.value
+
+
+def get_jetstream_pytorch_inference_nightly_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    num_slices: int = 1,
+    model_configs: Dict = {},
+):
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.BENCHMARK_DATASET,
+  )
+
+  set_up_cmds = (
+      "pip install --upgrade pip",
+      # Download jetstream and maxtext
+      "git clone https://github.com/google/jetstream-pytorch.git",
+      # Create a python virtual environment
+      "sudo apt-get -y update",
+      "sudo apt-get -y install python3.10-venv",
+      "sudo apt-get -y install jq",
+      "python -m venv .env",
+      "source .env/bin/activate",
+      # Setup MaxText & JetStream
+      "cd jetstream-pytorch && source install_everything.sh",
+  )
+
+  additional_metadata_dict = {
+      "checkpoint": f"{model_configs['checkpoint']}",
+      "dataset": f"{model_configs['dataset']}",
+      "max_output_length": f"{model_configs['max_output_length']}",
+      "batch_size": f"{model_configs['batch_size']}",
```

Comment:
We definitely want to write as much as possible to the metadata_dict, such as 
```
          "model_name": "llama-2",
          "size": "13b", 
          "checkpoint": "gs://inference-benchmarks/models/llama2-13b/pytorch/llama-2-13b/",
          "dataset": "openorca",
          "tokenizer": "tokenizer.llama2",
           batch_size
          "request_rate": 5,
          "num_prompts": 200,
          "max_output_length": 1024,
          "max_cache_length": 2048, 
          "sharding_config": "default_shardings/llama.yaml",
```
 

For example, you can see what we have for the Maxtext/Jetstream one. This PR has some more additions: https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/325/files#diff-db4e50a7951bdb1c5f9bd18530cc90fc87d09a1f70406106a41d4f48440b6653R70

Also for the `model_name` variable, we used the full name with size, such as `llama2-13b`. I see you split it up. Is it because the `convert_checkpoints` script requires `llama2` and `13b` separately? And also `benchmark_serving.py` https://github.com/google/JetStream/blob/main/benchmarks/benchmark_serving.py#L163 checks for `llama-3` rather than `llama-3-8b`.

Maybe we need to change the way we pass in `model_name` https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/325/files#diff-d8c4b4d0b14721ccb0806d911102e662d701984e5dd25e10cdd4092129a6563eR193

Reply:
`model_name` and `model_size` are two different fields when passed into `run_server` in jetstream-pytorch. I think we can keep the `test_models` keys in accord for the two different frameworks, but pass in model_name/size of different format.

Reply:
Can you give an example of the `test_models` key in this context?

Reply:
The `llama2-7b`, `llama2-70b` and `gemma-7b` in the maxtext tests, and `llama3-8b`, `llama2-7b` and `llama2-13b` in jetstream-pytorch